kythe/typescript/indexer.ts

```diff
@@ -885,16 +910,25 @@ class Visitor {
       // instances of the class) and a value (the constructor).
       const kClass = this.getSymbolName(sym, TSNamespace.TYPE);
       this.emitNode(kClass, 'record');
+      const classAnchor = this.newAnchor(decl.name);
+      this.emitEdge(classAnchor, 'defines/binding', kClass);
+
       const kClassCtor = this.getSymbolName(sym, TSNamespace.VALUE);
+      let classCtorAnchor;
+      const ctor = sym.members!.get(ts.InternalSymbolName.Constructor);
+      if (ctor) {
+        const decl = ctor.declarations[0];
+        const span = this.getTextSpan(decl, 'constructor');
+        classCtorAnchor = this.newAnchor(decl, span.start, span.end);
+      } else {
+        // TODO: the specific constructor() should really be the thing tagged
+        // with constructor, but we also need to handle classes that don't
+        // declare a constructor.  Fix me later.
```

Comment:
It's not clear what could be fixed in this case—if there is no constructor syntax, the choices are to choose some other syntax (here: the class declaration itself) or to omit the declaration entirely. Since the class declaration is the only thing that _has_ to be present, that seems like the only reasonable fallback.

In which case, I would suggest making that the default:

```ts
let classCtorAnchor = classAnchor;
if (ctor) {
   …
}
```

Reply:
Sorry, that comment was an artifact before this was fixed. The behavior added in this PR is what you described, so I'll get rid of the comment :-)